src/Uno.UI.Tests/Windows_UI_Xaml/Given_ThemeResource.cs

```diff
@@ -40,8 +43,155 @@ public void When_System_ThemeResource_Dark()
 			finally
 			{
 				//ensure the light theme is reset
-				app.ForceSetRequestedTheme(Windows.UI.Xaml.ApplicationTheme.Light);
+				await SwapSystemTheme();
 			}
 		}
+
+		[TestMethod]
+		public void When_App_ThemeResource_Default()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+			app.HostView.Children.Add(control);
+
+			Assert.AreEqual(Colors.LemonChiffon, (control.TestBorder.Background as SolidColorBrush).Color);
+		}
+
+		[TestMethod]
+		public void When_App_ThemeResource_Light()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+			app.HostView.Children.Add(control);
+
+			Assert.AreEqual(ApplicationTheme.Light, app.RequestedTheme);
+			Assert.AreEqual(Colors.RosyBrown, (control.TestBorder.BorderBrush as SolidColorBrush).Color);
+		}
+
+		[TestMethod]
+		public void When_Inherited_In_Template_Applied()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			control.InlineTemplateControl.ApplyTemplate();
+			control.TemplateFromResourceControl.ApplyTemplate();
+
+			var text2InlineBefore = control.InlineTemplateControl.TextBlock2.Text;
+			var text2ResourceTemplateBefore = control.TemplateFromResourceControl.TextBlock2.Text;
+
+			var text4InlineBefore = control.InlineTemplateControl.TextBlock4.Text;
+			var text4ResourceTemplateBefore = control.TemplateFromResourceControl.TextBlock4.Text;
+		}
+
+		[TestMethod]
+		public void When_Inherited_In_Template_Applied_XAML_Scope()
+		{
+			var app = UnitTestsApp.App.EnsureApplication();
+
+			var control = new Test_Control();
+
+			control.InlineTemplateControl.ApplyTemplate();
+			control.TemplateFromResourceControl.ApplyTemplate();
+
+			var text2InlineBefore = control.InlineTemplateControl.TextBlock2.Text;
+			var text2ResourceTemplateBefore = control.TemplateFromResourceControl.TextBlock2.Text;
+
+			var text4InlineBefore = control.InlineTemplateControl.TextBlock4.Text;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'text4InlineBefore'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)